src/index.cjs

```diff
@@ -98,12 +108,18 @@ class ScratchWebpackConfigBuilder {
             module: {
                 rules: [
                     {
-                        test: enableReact ? /\.[cm]?jsx?$/ : /\.[cm]?js$/,
+                        test: enableReact ?
+                            (enableTs ? /\.[cm]?[jt]sx?$/ : /\.[cm]?jsx?$/) :
+                            (enableTs ? /\.[cm]?[jt]s$/ : /\.[cm]?js$/),
                         loader: 'babel-loader',
                         exclude: [
                             {
                                 and: [/node_modules/],
-                                not: [/node_modules[\\/].*scratch/]
```

Comment:
That's a great find, thank you for looking into it in such detail. I'll update it to handle paint, render, svg-renderer and vm.

As a thought exercise, I'm wondering what the best option would be in non-monorepo case - whether packages should expose their source or the build output. I guess it depends on a lot of things, but the safest option is probably exposing the build output and using that in dependents, instead of the source. Because then each dependent needs to have a build configuration supporting the things the dependency uses.

Reply:
That's what we used to do: `scratch-vm` would consume the build output from `scratch-render`, for example. The problem was that we ended up having redundant content in build outputs. For example, `scratch-vm` and `scratch-render` both directly depend on `scratch-svg-renderer`, so `scratch-gui` would receive two "copies" of `scratch-svg-renderer` (one in the `scratch-vm` build output, and another in the `scratch-render` output).

I could think of two main solutions to this: either we could make all Scratch dependencies into externals, or we should build from source at the outermost level (`scratch-gui` / `scratch-www`). Maybe correctness would have been easier with externals, but I went for the "build from source" option because it seemed like it could be more optimal. That's the work I tried to do back in March and got stuck...

Premature optimization something something? Yeah, probably... :melting_face:

Reply:
It's definitely not clear-cut, depends on whether we view those sub-dependencies as libraries themselves, or just components that are kind-of but not really separate. In the monorepo case it's more clear win towards building from source since we can more easily share configuration and make sure we're not missing anything in terms of transforms.

Reply:
I changed the `not` clause here to `/node_modules[\\/]scratch-(paint|render|svg-renderer|vm)[\\/]src[\\/]/`, do you think that would handle all the cases above?